Automattic/jetpack
# Donations block: Remove conflicting styles with themes

Fixes https://github.com/Automattic/jetpack/issues/16749

#### Changes proposed in this Pull Request:

Removes the `wp-block-button_link` classes we were reusing for setting button styles for some elements in favor of fixed styles defined in the block stylesheets, making sure themes don't collide with them.

Theme | View | Before | After
--- | ---| --- | ---
Alves | Editor | <img width="841" alt="Screen Shot 2020-08-26 at 11 45 23" src="https://user-images.githubusercontent.com/1233880/91291216-03e98380-e795-11ea-973d-c6f3172eab81.png"> | <img width="817" alt="Screen Shot 2020-08-26 at 12 53 24" src="https://user-images.githubusercontent.com/1233880/91295305-27afc800-e79b-11ea-9476-b3d21ee9a687.png">
Alves | Frontend | <img width="787" alt="Screen Shot 2020-08-26 at 11 46 00" src="https://user-images.githubusercontent.com/1233880/91291298-2085bb80-e795-11ea-918e-4956865a72e4.png"> | <img width="795" alt="Screen Shot 2020-08-26 at 12 53 43" src="https://user-images.githubusercontent.com/1233880/91295325-326a5d00-e79b-11ea-9409-2e3a24067773.png">
Twenty Twenty | Editor | <img width="647" alt="Screen Shot 2020-08-26 at 12 43 28" src="https://user-images.githubusercontent.com/1233880/91295164-f1724880-e79a-11ea-9ab8-ab24b2686951.png"> | <img width="647" alt="Screen Shot 2020-08-26 at 12 51 04" src="https://user-images.githubusercontent.com/1233880/91295198-ff27ce00-e79a-11ea-9ab6-ced4c42cb4c0.png">
Twenty Twenty | Frontend | <img width="608" alt="Screen Shot 2020-08-26 at 12 43 45" src="https://user-images.githubusercontent.com/1233880/91295183-f931ed00-e79a-11ea-8056-fe7306410278.png"> | <img width="613" alt="Screen Shot 2020-08-26 at 12 51 48" src="https://user-images.githubusercontent.com/1233880/91295213-04851880-e79b-11ea-8d9b-b7b337e1dcb5.png">

#### Jetpack product discussion
pbMlHh-dW-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Set up a test site:
  * [Create a JN site running this branch](https://jurassic.ninja/create/?jetpack-beta&branch=update/donations-colors). 
  * Go to WP Admin > Settings > Jetpack Constants.
  * Set `JETPACK__SANDBOX_DOMAIN` with your sandbox address.
  * Sandbox the API and the store (see PCYsg-lW4-p2 #sandbox-method for detailed instructions).
  * Set up Jetpack and purchase a Premium or Professional plan.
* Download the Varia and Alves themes from here:
  * Varia: https://www.dropbox.com/s/iozerkwcn9wfizy/varia.zip?dl=0
  * Alves: https://www.dropbox.com/s/xyf3b285z5t0m27/alves.zip?dl=0
  * Alternatively, download them from the `master` branch of https://github.com/Automattic/themes. Do not use links available in https://wordpress.com/theme/ since they don't provide the latest version of the themes.
* Go to WP Admin > Appearance > Themes.
* Upload both of them, and activate Alves.
* Go to WP Admin > Posts > New.
* Insert a Donations block and connect to Stripe.
* Back in the editor, make sure the theme styles does not mess up the UI of the block.
* Publish the post and visit it.
* Make sure the theme styles does not mess up the UI of the block.

#### Proposed changelog entry for your changes:
Donations block: Add color settings
